Services/DataX.Gateway/DataX.Gateway.Api/Controllers/GatewayController.cs

```diff
@@ -153,7 +166,7 @@ public async Task<ApiResult> QueryService(HttpRequestMessage request, string app
         private Dictionary<string, string> FetchUserHeaders()
         {
             var headers = new Dictionary<string, string>();
-            var userName = ClaimsPrincipal.Current.FindFirst(ClaimTypes.Upn) != null ? ClaimsPrincipal.Current.FindFirst(ClaimTypes.Upn).Value : (ClaimsPrincipal.Current.FindFirst(ClaimTypes.Email)?.Value);
+            var userName = ClaimsPrincipal.Current.FindFirst(ClaimTypes.Upn) != null ? ClaimsPrincipal.Current.FindFirst(ClaimTypes.Upn).Value : ClaimsPrincipal.Current.FindFirst(ClaimTypes.Email)?.Value;
             headers.Add(_UserNameHeader, userName ?? string.Empty);
             var userId = ClaimsPrincipal.Current.FindFirst("http://schemas.microsoft.com/identity/claims/objectidentifier")?.Value;
```

Comment:
let use a combined id to identify a unique user. 
take a look at https://github.com/Azure-Samples/active-directory-dotnet-native-aspnetcore-v2/blob/63ee17b6dc0652eefb176672336f9f746d4ac01c/Microsoft.Identity.Web/ClaimPrincipalExtension.cs#L32

Reply:
the usage pattern is
```
var userId = ClaimsPrincipal.Current.GetMsalAccountId();
```

Reply:
For now I did not add another class but using the same pattern as you suggest here